### PR TITLE
use AtomicWaker instead of Mutex<Option<Waker>>

### PR DESCRIPTION
**Pull request by cfeitong**
`futures` crate provides a useful synchronization primitive `AtomicWaker`. I think it is a better choice over `Mutex<Option<Waker>>`.

### CODE REVIEW

**Comment by ibraheemdev** [comment]
Sorry for the late review, I don't want to pull in the entire futures crate for this, could you update the code to depend on the `atomic-waker` crate instead?

**Comment by honzchen** [comment]
@ibraheemdev friendly ping.

**Comment by ibraheemdev** [comment]
Sorry, I didn't get notified that you made the change. Just one small nit before merging.

File: src/lib.rs [hunk 1]
```diff
@@ -178,7 +179,7 @@ impl Drop for Worker {
         let count = self.inner.count.fetch_sub(1, Ordering::Relaxed);
         // We are the last worker
         if count == 1 {
-            if let Some(waker) = self.inner.waker.lock().unwrap().take() {
```
Comment (ibraheemdev) [inline src/lib.rs:181]: I think this can be replaced with just `self.inner.waker.wake()`

Comment (honzchen) [reply src/lib.rs:181]: > I think this can be replaced with just `self.inner.waker.wake()`

cc @cfeitong

Comment (ibraheemdev) [reply src/lib.rs:181]: Oh I thought you were the same person 😄 I'll just merge this and make the change myself.

**Comment by ibraheemdev** [comment]
Thanks!

**Pull request merged**